packages/common/infra/src/livedata/react.ts

```diff
@@ -3,13 +3,29 @@ import { useSyncExternalStore } from 'react';
 
 import type { LiveData } from './index';
 
+function noopSubscribe() {
+  return () => {};
+}
+
+function noopGetSnapshot() {
+  return null;
+}
+
 /**
  * subscribe LiveData and return the value.
  */
-export function useLiveData<T>(liveData: LiveData<T>): T {
+export function useLiveData<Input extends LiveData<any> | null | undefined>(
```

Comment:
in which scenario the LiveData passed to this hook is nullable?

Reply:
before:

```ts
const page = useServiceOptional(Page);
const pageId = page?.id;
const mode = useLiveData(page?.mode ?? new LiveData(null));
```

after:

```ts
const page = useServiceOptional(Page);
const pageId = page?.id;
const mode = useLiveData(page?.mode);
```

Reply:
page service is reactive means when page change, page service will change.

page is internally variable, page internal is not reactive